### ISSUE TITLE
crashes when trying to load in fl

### ISSUE THREAD

**Issue by coolkids94** [johnnovak/Nuked-SC55-CLAP#5]
it refuses to work in FL, it says the vst is an effect which is weird and trying to add it in just makes fruity wrapper crash, i have put the clap in the clap folder and got the right roms for it

**Comment by johnnovak** [comment]
> it refuses to work in FL, it says the vst is an effect which is weird and trying to add it in just makes fruity wrapper crash, i have put the clap in the clap folder and got the right roms for it

I haven't uses Fruity Loops in ages 😄 Does it support CLAP natively or are you using a 3rd partt wrapper? Are you on Windows? Can you get other CLAP instruments work in it?

**Comment by coolkids94** [comment]
the newest versions of now support CLAP natively, im on windows and other CLAP instruments work just fine

**Comment by johnnovak** [comment]
I will test and fix it at some point, but I'm busy with other stuff. Probably next year.

**Comment by fotisandstuff** [comment]
It also crashes for me. I use Reaper (which natively supports CLAP) on Win 10 x64.

**Comment by johnnovak** [comment]
> It also crashes for me. I use Reaper (which natively supports CLAP) on Win 10 x64.

It works in REAPER. You must have screwed up the ROM file placement.

**Comment by fotisandstuff** [comment]
Sorry, my previous post was wrong. I meant to say that Reaper cannot even scan the plugin.
I have placed the file `Nuked-SC55.clap` in the folder `c:\Program Files\Common Files\Clap\`
As for the `Nuked-SC55-Resources`, I grabbed it from the link you provided and copied the folder in `c:\Program Files\Common Files\Clap\` as well (so it's `c:\Program Files\Common Files\Clap\Nuked-SC55-Resources`). I have kept all subfolders intact.
Still, Reaper fails to scan the plugin. I have cleared the cache and re-scanned all CLAP files repeatedly, but still nothing.

**Comment by johnnovak** [comment]
@fotisandstuff Using 64-bit REAPER?

**Comment by johnnovak** [comment]
Not sure what you're doing wrong, but here's the proof it's working:

Inside `C:\Program Files`

![Image](https://github.com/user-attachments/assets/d5ce31d5-3c0a-42ca-8459-83795907d780)

After clear & rescan:

![Image](https://github.com/user-attachments/assets/eac4d40f-6afd-42b3-b304-1db4f3f92edf)

![Image](https://github.com/user-attachments/assets/f40c85cf-f648-454b-b431-7da3d3cb1f37)

![Image](https://github.com/user-attachments/assets/ee35420e-69d2-4a17-b2c3-389aa60331fe)

**Comment by fotisandstuff** [comment]
![Image](https://github.com/user-attachments/assets/540ab146-1fa8-4bb7-b31c-0691bb797a0b)

![Image](https://github.com/user-attachments/assets/87b66158-1b11-46e1-a488-0986091b9630)

![Image](https://github.com/user-attachments/assets/60e7f348-ef2c-4643-889b-f2a1d40023b2)

**Comment by johnnovak** [comment]
@fotisandstuff I can try REAPER 7.30 but I think it's something else... 🤷🏻

**Comment by tomcita** [comment]
Same problem here. I tried on Reaper 7.30, 7.31 and 7.33, same error: plugin failed to scan

**Comment by johnnovak** [comment]
I've just upgraded to REAPER 7.33 and have deleted the plugin then reinstalled it from scratch using the release download and the instructions from the README.

I'm listening to the DOOM level 1 music in REAPER while typing this 🤷🏻 

So, user error. Maybe check the file permissions. I can't fix it until you tell me exactly what's wrong, and I can't trigger the problems. Works fine in bot 64 and 32-bit versions of REAPER 🤷🏻

**Comment by marsej** [comment]
I have the same problem. Here's what I have done so far:
I started with clean vmware Windows 10 guest install (same issue on Win10 host). 

Installed VS2022 and Reaper 7.33 on the guest. In VS2022 install I selected the latest Win10 platform SDK (not win11)
I tried to install vc_redist.x64.exe 2015-2022 C++ redistributable libraries but it says I already have them (presumably from installing VS 2022). 

(side-step I did first: I downloaded micro-clap-host-main and built it and the Nuked-SC55-Clap for debugging. I thought I had the same problem with this but started to doubt it after I got the micro-clap-host to load the plugin and print some info)

I tried to load the debug dll into Reaper without success. Then I tried attaching to Reaper but Reaper loads the plugin through starting another instance that it uses probably pipe to communicate with. (reaper nov 2020 change "VST: use separate process when scanning plug-ins, allow terminating hung plug-ins during scan")

I used sysmon from Microsoft to find out how Reaper does the plugin scan and tip from SO to use gflags on Reaper to see what the library loading process looks like. The last parameter ("AFH0lqhN2wFbtcbOOn/bAQCHFAA=") is plugin-specific. I also tried another free CLAP plugin (JC303) which loaded fine without the ERROR:'s below.

"C:\Program Files (x86)\Windows Kits\10\Debuggers\x64\gflags.exe" -i reaper.exe +sls

part of windbg output:

CommandLine: "C:\Program Files\REAPER (x64)\reaper.exe" "-cfgfile" "C:\Users\Admin\AppData\Roaming\REAPER\REAPER.ini" "-__vst_scan__" "C:\Program Files\Common Files\CLAP\John Novak\Nuked-SC55\Nuked-SC55.clap" "AFH0lqhN2wFbtcbOOn/bAQCHFAA="

283c:13a4 @ 73497515 - LdrpProcessWork - ERROR: Unable to load DLL: "libspeexdsp.dll", Parent Module: "C:\Program Files\Common Files\CLAP\John Novak\Nuked-SC55\Nuked-SC55.clap", Status: 0xc0000135
283c:3098 @ 73497515 - LdrpProcessWork - ERROR: Unable to load DLL: "VCRUNTIME140.dll", Parent Module: "C:\WINDOWS\SYSTEM32\MSVCP140.dll", Status: 0xc0000135
ModLoad: 00000000`00440000 00000000`0045e000   C:\WINDOWS\SYSTEM32\VCRUNTIME140.dll


Now I'm stuck/out of ideas. I have VCRUNTIME140.dll in SYSTEM32 and the libspeexdsp.dll in same folder as the .clap. Yet I get these errors.

**Comment by johnnovak** [comment]
@marsej Speex is statically linked to the plugin if you follow the build instructions. Don't know how you're building it, but doesn't seem like you're following the instructions.

**Comment by marsej** [comment]
After removing the libspeexdsp.dll, I started getting LoadLibrary Error 126. This made me think the problem may be in speexdsp.lib.
Then I (temporarily, just to see what happens) removed all referenced to Speex from the plugin source code and the linker options and now the LoadLibrary works and Reaper finds the plugins.

I guess next move is to look at what is going on in the making of the speexdsp.lib.

**Comment by marsej** [comment]
Replaced
vcpkg_installed\x64-windows\(release/debug)\lib\speexdsp.lib
with unofficial build
C:\vs2022repos\speexdsp-win64-SpeexDSP-1.2.0\lib\libspeexdsp.lib
from https://github.com/gavv/speexdsp-win64

and now Reaper finds the plugin. However something is not quite right, I tried loading the different ROM plugins and all but sc-55mkII have some sort of inaudible sound that causes the Reaper volume meter to go -12dB and there's pops etc.

**Comment by coolkids94** [comment]
i acidentally closed the issue, apologies for that but so it seems like FL doesnt scan it cuz of something broke in that speex library?

**Comment by marsej** [comment]
As said, I had some success in Reaper with the plugin (piano sound) using the other lib statically linked but in Fruity Loops I can get sound from JC303 clap plugin but not from Nuked-SC55-Clap. So there may still be work to be done on getting it to work in at least Fruity Loops.

**Comment by 0mnicydle** [comment]
I had this same issue and fixed it by specifying the correct triplet for the cmake config:

`cmake -DVCPKG_TARGET_TRIPLET=x64-windows-static-md -G "Visual Studio 17 2022" --preset=ninja`

From what I gather, `set(VCPKG_LIBRARY_LINKAGE static)` is only valid in a triplet definition, and I don't think it's actually doing anything as used in CMakeLists.txt here, but I could be wrong... In any case, above command allowed me to build the static version.

**Comment by marsej** [comment]
^That solved the linkage problem.

I still get pop-sounds when toggling the non-mk2 CLAPi's in Reaper and also when playing a sound with the mk2 CLAPi enabled while simultaneously starting up Fruity Loops in vmware player win10 guest. There's no popping in same scenario with Munt and OsTIrus, which both use ROM's also.

Also in Fruity Loops, there's no error when adding the SC-55(mk2) CLAPi but FL says it's a "sampler" and the midi i/o etc cannot be configured. JC303 loads correctly as a synth and has midi i/o menu.

**Comment by johnnovak** [comment]
@0mnicydle Thanks for the tip about static linking. I wasn't aware vcpkg defaults to dynamic linking on Windows (on macOS and Linux, it defaults to static linking).

I've fixed up the README and I've reuploaded the Windows release artifacts. Let me know if this works for you now (@0mnicydle @fotisandstuff @marsej).

**Comment by 0mnicydle** [comment]
> Let me know if this works for you now

Sadly not. I think there's multiple issues here. When I try to build the clap now I get:

```
cl : command line  error D8021: invalid numeric argument '/Wextra' [D:\Temp\git\Nuked-SC55-CLAP\build\Nuked-SC55-CLAP.v
cxproj]
```

My best guess is nothing has set CMAKE_SYSTEM_NAME STREQUAL by the time it's referenced in CmakeLists.txt so I'm getting the macos/linux compile options added to the project. If I manually set CMAKE_SYSTEM_NAME STREQUAL to "windows" it then uses the triplet line, but then cmake fails because:

`error: Invalid triplet: x64-`

So looks like there might be a syntax error in the triplet line.

**Comment by 0mnicydle** [comment]
Updated my last comment, second problem was my own fault.

**Comment by 0mnicydle** [comment]
Looks like you may want to use `CMAKE_HOST_SYSTEM_NAME` instead.

> Note that CMAKE_SYSTEM_NAME is not set to anything by default when running in script mode, since it's not building anything.

**Comment by fotisandstuff** [comment]
It's working!!! All four CLAPs now show up in Reaper and make sounds <3

**Comment by 0mnicydle** [comment]
I was thinking more about this and it occurred to me using CMAKE_HOST_SYSTEM_NAME probably isn't desired as I'm guessing it could interfere with cross-compiling...

**Comment by 0mnicydle** [comment]
I did some more digging and determined CMAKE_SYSTEM_NAME doesn't get populated until after the project command, however setting the triplet variable seems to needed before the project command :D

**Comment by johnnovak** [comment]
@0mnicydle Yeah I couldn't figure it either how to set this only for Windows is CMakeList.txt. I just compiled it in the end with the command line you posted first, then went to bed because I was really tired 😄 

I think putting the flag into some new Windows CMake presets would work.

**Comment by 0mnicydle** [comment]
> I think putting the flag into some new Windows CMake presets would work.

I wanted to suggest that but didn't want to come across as defeatist if there was a better solution :D

**Comment by 0mnicydle** [comment]
@marsej

> I still get pop-sounds when toggling the non-mk2 CLAPi's in Reaper

The popping sounds with the mk1 roms is because for some reason these are always outputting noise even when idle. I confirmed this happens with the standalone Nuked SC-55 app too by piping the audio into reaper. If this is an issue it's an upstream one, but for all I know this is accurate behavior for a real mk1 SC-55.

![Image](https://github.com/user-attachments/assets/c1bc5308-6b19-4045-8832-45549d7c4465)

**Comment by johnnovak** [comment]
> The popping sounds with the mk1 roms is because for some reason these are always outputting noise even when idle. I confirmed this happens with the standalone Nuked SC-55 app too by piping the audio into reaper. If this is an issue it's an upstream one, but for all I know this is accurate behavior for a real mk1 SC-55.

Yeah, my real hardware mk1 does that weird background noise thing, so it's accurate emulation. It's a bit random, depending on what MIDI events you sent the device beforehands. 

This annoyed me when I automatically recorded 50+ game soundtrack MIDIs... I just put tge MIDI files one after other, with a 10 second or so gap between them, then hit record in REAPER and let it run for 1.5 hour or so.

Well, there was a quite audible noise in those 10 second gaps between _some_ tunes if you turned up the volume. So I had to re-record a few, and on the second attempt the static bacmground noise was gone. So it' srate-dependent and kinda semi-random how loud it gets.

> Wait, you are getting that orange GUI when loading the plugin in Reaper? I get GUI with other plugins but not with Nuked SC-55 CLAP.

@marsej Lol, my CLAP plugin has no UI. That screenshot only illustrated the issue is present in original Nuked SC-55. @0mnicydle said so:

>  I confirmed this happens with the standalone Nuked SC-55 app too by piping the audio into reaper.
